### PR TITLE
Replace DS Promises with RSVP Promise in Session model.

### PR DESCRIPTION
**Pull request by stopfstedt**
refs #2034

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit 0e7fcbdf1077e646c2c1f5d071ec1783ad4a276b

https://deploy-preview-2771--ilios-frontend.netlify.com

File: app/models/session.js [hunk 1]
```diff
@@ -6,132 +6,126 @@ import CategorizableModel from 'ilios/mixins/categorizable-model';
...
+  /**
+   * All offerings for this session, sorted by offering startdate in ascending order.
+   * @property sortedOfferingsByDate
+   * @type {Ember.computed}
+   */
+  sortedOfferingsByDate: computed('offerings.@each.startDate', function() {
+    return new Promise(resolve => {
       this.get('offerings').then(offerings => {
-        let sortedOfferingsByDate = offerings.filter(offering => isPresent(offering.get('startDate'))).sort((a, b) => {
+        resolve(offerings.filter(offering => isPresent(offering.get('startDate'))).sort((a, b) => {
```
Comment (jrjohnson) [inline app/models/session.js]: I think wrapping this in a resolve gets a little bit hard to read.  Might be easier to read if it assigned to var, filtered and assigned to var, then sorted and assigned, then resolved.  Is that too verbose?

Comment (stopfstedt) [inline app/models/session.js]: nah, you're right. will make it a bit less dense.

Comment (stopfstedt) [inline app/models/session.js]: handled.

**Pull request merged**